polaris/evaluate/_predictions.py

```diff
@@ -177,3 +188,35 @@ def _is_fully_specified(
                 return False
 
         return True
+
+
+class CompetitionPredictions(BenchmarkPredictions, BaseArtifactModel):
+    """
+    Predictions for competition benchmarks.
+
+    This object is to be used as input to
+    [`PolarisHubClient.submit_competition_predictions`][polaris.hub.client.PolarisHubClient.submit_competition_predictions].
+    It is used to ensure that the structure of the predictions are compatible with evaluation methods on the Polaris Hub.
+    In addition to the predictions, it contains meta-data that describes a predictions object.
+
+    Attributes:
+        name: A slug-compatible name for the artifact. It is redeclared here to be required.
+        owner: A slug-compatible name for the owner of the artifact. It is redeclared here to be required.
+        report_url: A URL to a report/paper/write-up which describes the methods used to generate the predictions.
+        github_url: A URL to a code repository which contains the code used to generate the predictions.
+        contributors: A list of Polaris usernames representing those who contributed to producing the predictions.
+    """
+
+    _artifact_type = "competition-prediction"
+
+    name: SlugCompatibleStringType
+    owner: HubOwner
```

Comment:
Could you see people using the `CompetitionPredictions` object for something other than uploading their predictions to the Hub? There isn't much more functionality within the class that users can leverage. 

I'm not against making the `name`, `owner` and `report_url` field optional on the client and mandatory on the server. In my view, it just depends on if we think there is a use case for that object pre-upload.

Reply:
I don't feel strongly either way here. I mostly comment because I do think it's important to be consistent if we can to simplify maintenance. 

Bit of a tangent, but I think the UX for validation ahead of upload can be improved. The Zod errors we now surface to the client are rather cryptic.

Reply:
Agreed. The cryptic errors is one, and the inconsistent formatting/highlighting of error messages can be improved too. It would be really nice to create a package for updating this messaging in the new year.